### PR TITLE
Add CREATE TABLE AS to Postgres and Redshift

### PR DESCRIPTION
**Pull request by tdstark**
### Summary
This PR fixes issue #1781 by adding CREATE TABLE AS into the Postgres dialect. These changes will be inherited by Redshift dialect. In addition, test cases were added to check for compatibility with and without parenthesis surrounding the select statement.

In the future, `TABLE`, `VALUES`, AND `EXECUTE` statements will need to be added as options along with `SELECT`. However, this will likely cover the bulk of use cases for now.

### Are there any other side effects of this change that we should be aware of?
At this time, there are no known side effects.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_postgres.py [hunk 1]
```diff
@@ -932,6 +932,77 @@ class CreateTableStatementSegment(BaseSegment):
...
+                "COMMIT",
+                OneOf("PRESERVE ROWS", "DELETE ROWS", "DROP"),
+                optional=True,
+            ),
+            Sequence("TABLESPACE", Ref("ParameterNameSegment"), optional=True),
+        ),
+        "AS",
+        OneOf(
+            OptionallyBracketed(Ref("SelectableGrammar"))
+            # TODO Add TABLE, VALUES, AND EXECUTE commands
```
Comment (tunetheweb) [inline src/sqlfluff/dialects/dialect_postgres.py:1000]: These might be as easy as this to add:

```suggestion
            Sequence("TABLE", Ref("TableReferenceSegment"),
            Ref("ValuesClauseSegment"),
            Sequence("EXECUTE", Ref("FunctionSegment")),
```

Though should add some test examples if willing to add this.

File: src/sqlfluff/dialects/dialect_postgres.py [hunk 2]
```diff
@@ -932,6 +932,77 @@ class CreateTableStatementSegment(BaseSegment):
...
+    """
+
+    type = "create_table_as_statement"
+
+    match_grammar = Sequence(
+        "CREATE",
+        OneOf(
+            Sequence(
+                OneOf("GLOBAL", "LOCAL", optional=True),
+                Ref("TemporaryGrammar", optional=True),
```
Comment (WittierDinosaur) [inline src/sqlfluff/dialects/dialect_postgres.py:949]: This optional shouldn't be true. You can't have global/local without temporary grammar.

File: src/sqlfluff/dialects/dialect_postgres.py [hunk 3]
```diff
@@ -932,6 +932,77 @@ class CreateTableStatementSegment(BaseSegment):
...
+        "TABLE",
+        Ref("IfNotExistsGrammar", optional=True),
+        Ref("TableReferenceSegment"),
+        AnyNumberOf(
+            Sequence(
+                Bracketed(
+                    Delimited(
+                        Sequence(
+                            Ref("ColumnReferenceSegment"),
+                            AnyNumberOf(Ref("ColumnConstraintSegment")),
```
Comment (WittierDinosaur) [inline src/sqlfluff/dialects/dialect_postgres.py:963]: I don't think CREATE TABLE AS supports column constraints, just renaming of the column name

File: src/sqlfluff/dialects/dialect_postgres.py [hunk 4]
```diff
@@ -932,6 +932,77 @@ class CreateTableStatementSegment(BaseSegment):
...
+            ),
+            Sequence("USING", Ref("FunctionSegment"), optional=True),
+            OneOf(
+                Sequence(
+                    "WITH",
+                    Bracketed(
+                        AnyNumberOf(
+                            Sequence(
+                                Ref("ParameterNameSegment"),
+                                Sequence(
```
Comment (WittierDinosaur) [inline src/sqlfluff/dialects/dialect_postgres.py:977]: This second nested sequence isn't needed, just put those terms in the one above

File: src/sqlfluff/dialects/dialect_postgres.py [hunk 5]
```diff
@@ -932,6 +932,77 @@ class CreateTableStatementSegment(BaseSegment):
...
+                        )
+                    ),
+                    optional=True,
+                ),
+                Sequence("WITHOUT", "OIDS", optional=True),
+            ),
+            Sequence(
+                "ON",
+                "COMMIT",
+                OneOf("PRESERVE ROWS", "DELETE ROWS", "DROP"),
```
Comment (WittierDinosaur) [inline src/sqlfluff/dialects/dialect_postgres.py:992]: PRESERVE ROWS and DELETE ROWS should be sequences

File: src/sqlfluff/dialects/dialect_postgres.py (hunk 1 above)
Comment (WittierDinosaur) [reply src/sqlfluff/dialects/dialect_postgres.py:1000]: I think this is worth adding in this PR, as it's not a complicated change

**Review by WittierDinosaur — CHANGES_REQUESTED** [review]
Looks good for the most part. See comments for changes. Also, it'd be good to add a bunch more test cases for the various parts of the statement

File: src/sqlfluff/dialects/dialect_postgres.py (hunk 2 above)
Comment (tdstark) [reply src/sqlfluff/dialects/dialect_postgres.py:949]: @WittierDinosaur This is replicating what exists in `CreateTableStatement`. Would the following work?: 
```
        OneOf(
            Ref("TemporaryGrammar"),
            Sequence(
                OneOf("GLOBAL", "LOCAL"),
                Ref("TemporaryGrammar"),
            ),
            "UNLOGGED",
            optional=True,
        ),
```

File: src/sqlfluff/dialects/dialect_postgres.py (hunk 4 above)
Comment (tdstark) [reply src/sqlfluff/dialects/dialect_postgres.py:977]: @WittierDinosaur This is also replicating what exists in the `CreateTableSegmentStatement`. I can remove it but the logic is the same according to the documentation.

File: src/sqlfluff/dialects/dialect_postgres.py (hunk 1 above)
Comment (tdstark) [reply src/sqlfluff/dialects/dialect_postgres.py]: This has been added.

File: src/sqlfluff/dialects/dialect_postgres.py (hunk 3 above)
Comment (tdstark) [reply src/sqlfluff/dialects/dialect_postgres.py]: Agree. This has been changed.

File: src/sqlfluff/dialects/dialect_postgres.py (hunk 2 above)
Comment (WittierDinosaur) [reply src/sqlfluff/dialects/dialect_postgres.py]: All you need to do is remove the optional=True from Ref(TemporaryGrammar), leave it in for OneOf(GLOBAL, LOCAL)

File: src/sqlfluff/dialects/dialect_postgres.py (hunk 4 above)
Comment (WittierDinosaur) [reply src/sqlfluff/dialects/dialect_postgres.py:973]: It's probably unnecessary if CreateTable too. For clarity, I'm just talking about de-nesting the inner sequence, not getting rid of what it contains. Bring the EqualsSegment and LiteralGrammar to the same level as ParameterNameSegment

File: src/sqlfluff/dialects/dialect_postgres.py (hunk 2 above)
Comment (tdstark) [reply src/sqlfluff/dialects/dialect_postgres.py]: Done.

File: src/sqlfluff/dialects/dialect_postgres.py (hunk 4 above)
Comment (tdstark) [reply src/sqlfluff/dialects/dialect_postgres.py]: It should be removed from both now. It didn't seem to have any affect on the tests.

File: src/sqlfluff/dialects/dialect_postgres.py [hunk 6]
```diff
@@ -911,11 +911,9 @@ class CreateTableStatementSegment(BaseSegment):
...
                             Sequence(
                                 Ref("ParameterNameSegment"),
-                                Sequence(
-                                    Ref("EqualsSegment"),
-                                    Ref("LiteralGrammar"),
-                                    optional=True,
-                                ),
+                                Ref("EqualsSegment"),
+                                Ref("LiteralGrammar"),
+                                optional=True,
```
Comment (WittierDinosaur) [inline src/sqlfluff/dialects/dialect_postgres.py:916]: I've just reread the docs, and realised that asking you to change this here and in createtableas was bad advice 🤦‍♂️ . Can you change it back? And then I think we're good to go.

Comment (tdstark) [reply src/sqlfluff/dialects/dialect_postgres.py]: Done!

**Review by WittierDinosaur — APPROVED** [review]

**Pull request merged**